include/revng/Model/ConvertFunctionType.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Type.h"
+
+namespace model {
+
+/// Best effort `CABIFunctionType` to `RawFunctionType` conversion.
+std::optional<model::CABIFunctionType>
+convertToCABIFunctionType(const model::RawFunctionType &Function,
+                          model::abi::Values ABI,
+                          model::Binary &TheBinaryToAddSupportTypesTo);
+
+/// Best effort `RawFunctionType` to `CABIFunctionType` conversion.
+///
+/// \note: this convertion is loosy since there's no way to represent some types
+/// in `RawFunctionType` in a reversible manner.
+std::optional<model::RawFunctionType>
+convertToRawFunctionType(const model::CABIFunctionType &Function,
+                         model::Binary &TheBinaryToAddSupportTypesTo);
+
+namespace detail {
+
+class DescendedFunction;
+
+} // namespace detail
+
+/// Allows to temporarily represent a `CABIFunctionType` object as
+/// a `RawFunctionType` allowing modifications that can later be propagated
+/// back to the original.
+///
+/// \note: is still unfinished, for now it only does the best effort convertion.
+inline detail::DescendedFunction
+modifyAsRawFunctionType(model::CABIFunctionType &Function,
+                        model::Binary &TheBinaryToAddSupportTypesTo);
+
+namespace detail {
+
+/// Serves as a RAII object for a temporary "double" function type
+/// representation.
+class DescendedFunction {
```

Comment:
This is too WIP to be merged, let's postpone it.

Reply:
Gone.